src/NSwag.SwaggerGeneration.AspNetCore/OperationResponseProcessor.cs

```diff
@@ -0,0 +1,222 @@
+                {
+                    var httpStatusCode = statusCodeGroup.Key;
+                    var returnType = statusCodeGroup.Select(r => r.ResponseType).FindCommonBaseType();
+                    var description = string.Join("\nor\n", statusCodeGroup.Select(r => r.Description));
+
+                    var typeDescription = _settings.ReflectionService.GetDescription(
+                        returnType, context.MethodInfo.ReturnParameter?.GetCustomAttributes(), _settings);
+                    var response = new SwaggerResponse
+                    {
+                        Description = description ?? string.Empty
+                    };
+
+                    if (IsVoidResponse(returnType) == false)
+                    {
+                        response.IsNullableRaw = responses.Any(r => r.IsNullable) && typeDescription.IsNullable;
+                        response.ExpectedSchemas = await GenerateExpectedSchemasAsync(context, statusCodeGroup);
+                        response.Schema = await context.SchemaGenerator
+                            .GenerateWithReferenceAndNullability<JsonSchema4>(
+                                returnType, null, typeDescription.IsNullable, context.SchemaResolver)
+                            .ConfigureAwait(false);
+                    }
+
+                    context.OperationDescription.Operation.Responses[httpStatusCode] = response;
+                }
+
+                var operationResponses = context.OperationDescription.Operation.Responses;
+                var hasSuccessResponse = operationResponses.ContainsKey("200") || operationResponses.ContainsKey("204");
+
+                var loadDefaultSuccessResponseFromReturnType = !hasSuccessResponse &&
+                    context.MethodInfo.GetCustomAttributes()
+                        .Any(a => a.GetType().IsAssignableTo("SwaggerDefaultResponseAttribute", TypeNameStyle.Name)) ||
+                    context.MethodInfo.DeclaringType.GetTypeInfo().GetCustomAttributes()
+                        .Any(a => a.GetType().IsAssignableTo("SwaggerDefaultResponseAttribute", TypeNameStyle.Name));
+
+                if (loadDefaultSuccessResponseFromReturnType)
+                    await LoadDefaultSuccessResponseAsync(context.OperationDescription.Operation, context.MethodInfo,
+                        successXmlDescription, context.SchemaGenerator, context.SchemaResolver).ConfigureAwait(false);
+            }
+            else
+                await LoadDefaultSuccessResponseAsync(context.OperationDescription.Operation, context.MethodInfo,
+                    successXmlDescription, context.SchemaGenerator, context.SchemaResolver).ConfigureAwait(false);
+
+            return true;
+        }
+
+        private async Task<ICollection<JsonExpectedSchema>> GenerateExpectedSchemasAsync(OperationProcessorContext context, IGrouping<string, OperationResponseModel> group)
+        {
+            if (group.Count() > 1)
```

Comment:
I don't think this check is necessary, I think a grouping always has at least one member.